pkg/pillar/cipher/cipher.go

```diff
@@ -34,13 +34,44 @@ func getEncryptionBlock(
 // GetCipherCredentials : decrypt encryption block
 func GetCipherCredentials(ctx *DecryptCipherContext,
 	status types.CipherBlockStatus) (types.CipherBlockStatus, types.EncryptionBlock, error) {
+	var decBlock types.EncryptionBlock
+	cipherBlock, clearBytes, err := GetCipherMarshalledData(ctx, status)
+	if err != nil {
+		return handleCipherBlockCredError(ctx, &cipherBlock,
+			decBlock, err, types.DecryptFailed)
+	}
+
+	var zconfigDecBlock zcommon.EncryptionBlock
+	err = UnmarshalCipherData(ctx, clearBytes, &zconfigDecBlock)
+	if err != nil {
+		ctx.Log.Errorf("%s, encryption block unmarshall failed, %v\n",
+			cipherBlock.Key(), err)
+		return handleCipherBlockCredError(ctx, &cipherBlock,
+			decBlock, err, types.UnmarshalFailed)
+	}
+	ctx.Log.Functionf("%s, cipherblock decryption successful", cipherBlock.Key())
+	decBlock = getEncryptionBlock(&zconfigDecBlock)
+	ctx.AgentMetrics.RecordSuccess(ctx.Log)
+	return cipherBlock, decBlock, err
+}
+
+// GetCipherMarshalledData : decrypt encryption block, without assuming the proto
+// encoding data is zcommon.EncryptionBlock, since the data before encryption may have
+//
+//	its own data structure, it is up to the caller to unmarshal the data using
```

Comment:
the formatting is a bit off

Reply:
fixed. it was a 'tab' instead of space in the front, caused the 'go fmt' to generate two empty lines.